### DISCUSSION TITLE
[Issue] Switch to Binance & Binance Testnet Network with custom RPC configuration

### DISCUSSION THREAD

**Discussion by Sl00x** [wevm/wagmi#450] [General]
### [Issue] Switch to Binance & Binance Testnet Network with custom RPC configuration 
<img width="885" alt="image" src="https://user-images.githubusercontent.com/30964327/168036276-79fe51a4-7667-40e1-a4ad-1b2d501cd8c3.png">

This configuration is working and when I log rpcUrl and chain it shows the right one !
But when I'm using
`const provider = useProvider()`
it returns this:
`_network: {
    name: 'homestead',
    chainId: 1,
    ensAddress: '0x00000000000C2E074eC69A0dFb2997BA6C7d2e1e',
    _defaultProvider: [Function: func] { renetwork: [Function (anonymous)] }
  },`

Have you got any ideas, How to configure binance testnet to use with wagmi ?

**Comment by Trashimi** [comment]
I've been kind of running into the same issue, +1 for this waiting for updates :)

**Comment by jxom** [comment]
We just merged in #408 which should fix RPC configuration issues. We will be releasing this soon, so hopefully this will fix your issue.

For the meantime, is your `provider` on `createClient` chain-aware? Can you provide an example of where you placed your `binanceTestnetChain`?

**Reply by Sl00x** [reply]
my app.tsx 
<img width="1079" alt="image" src="https://user-images.githubusercontent.com/30964327/168040452-f0181a40-a583-49b4-b29f-41d61f66d61c.png">


My wagmiConnector including Binance config:
<img width="1079" alt="image" src="https://user-images.githubusercontent.com/30964327/168040638-e9ba5027-6da6-414a-a744-6027d3890bfa.png">

**Comment by jxom** [comment]
With the introduction of `0.9.x`, wagmi now comes with the `bsc` & `bscTestnet` chains built-in (re-exported from our [`@wagmi/chains` reference repository](https://github.com/wagmi-dev/references/tree/main/packages/chains)). Read more [here](https://wagmi.sh/react/chains).

**Comment by cngvc** [comment]
In my case, the error comes from RPC url in metamask setting and `bscTestnet` object is not the same because I used [https://chainlist.org/chain/97](Chainlist) to add network.
`import { bscTestnet } from "wagmi/chains";`

Solution:
- Change network RPC in metamask config to `https://data-seed-prebsc-1-s1.binance.org:8545`
